jgrapht-core/src/main/java/org/jgrapht/graph/DirectedAcyclicGraph.java

```diff
@@ -15,71 +15,61 @@
  * (b) the terms of the Eclipse Public License v1.0 as published by
  * the Eclipse Foundation.
  */
-package org.jgrapht.experimental.dag;
+package org.jgrapht.graph;
 
 import java.io.*;
 import java.util.*;
 
 import org.jgrapht.*;
-import org.jgrapht.graph.*;
 import org.jgrapht.traverse.*;
 
 /**
+ * A directed acyclic graph (DAG).
+ * 
  * <p>
- * DirectedAcyclicGraph implements a DAG that can be modified (vertices &amp; edges added and
- * removed), is guaranteed to remain acyclic, and provides fast topological order iteration.
- * </p>
+ * Implements a DAG that can be modified (vertices &amp; edges added and removed), is guaranteed to
```

Comment:
The running time is a bit difficult to state in a simple way without ending up copying the whole intro from the paper. Nevertheless, I added one sentence and pointed the user to the paper.

Regarding which one is better, there is no clear winner. For a single edge addition (and sparse graph) it will probably be better to use the DAG. However, if someone can collect a batch of edges and add them at once, using `TopologicalOrderIterator` should amortize better.